lib/valkyrie/specs/shared_specs/supports_ordered_persisting.rb

```diff
@@ -14,12 +15,64 @@ class CustomResource < Valkyrie::Resource
   subject { persister }
   let(:resource_class) { CustomResource }
   let(:resource) { resource_class.new }
-  it "saves ordered properties and returns them in the appropriate order" do
-    resource.authors = ["a", "b", "a"]
+
+  it "orders string values and returns them in the appropriate order" do
+    validate_order ["a", "b", "a"]
+  end
+
+  it "orders boolean values and returns them in the appropriate order" do
+    validate_order [true, false, true]
+  end
+
+  it "orders integer values and returns them in the appropriate order" do
+    validate_order [1, 2, 1]
+  end
+
+  it "orders date values and returns them in the appropriate order" do
+    now = Time.now.round(3).utc
+    validate_order [now, now - 3.hours, now - 1.hour]
+  end
+
+  it "orders URIs and returns them in the appropriate order" do
+    uri1 = RDF::URI("http://example.com/foo")
+    uri2 = RDF::URI("http://example.com/bar")
+    uri3 = RDF::URI("http://example.com/baz")
+    validate_order [uri1, uri2, uri3]
+  end
+
+  it "orders IDs and returns them in the appropriate order" do
+    page1 = persister.save(resource: resource_class.new(authors: ["Page 1"]))
+    page2 = persister.save(resource: resource_class.new(authors: ["Page 2"]))
+    page3 = persister.save(resource: resource_class.new(authors: ["Page 3"]))
+    validate_order [page1.id, page2.id, page3.id]
+  end
+
+  it "orders floating point values and returns them in the appropriate order" do
+    validate_order [1.123, 2.222, 1.123]
+  end
+
+  # does not work in any persister
+  xit "orders nested objects" do
+    nested1 = resource_class.new(id: Valkyrie::ID.new("resource1"), authors: ["Resource 1"])
+    nested2 = resource_class.new(id: Valkyrie::ID.new("resource2"), authors: ["Resource 2"])
+    nested3 = resource_class.new(id: Valkyrie::ID.new("resource3"), authors: ["Resource 3"])
+    values = [nested1, nested2, nested3]
+
+    resource.nested = values
+
+    output = persister.save(resource: resource)
+    expect(output.authors).to eq(values)
```

Comment:
You set the `nested` attribute, not the `authors` attribute, and I don't think you can compare different copies of `Valkyrie::Resources` to one another.